### DISCUSSION TITLE
How to change RGB color in bulk?

### DISCUSSION THREAD

**Discussion by Python-Zhou** [ImageMagick/ImageMagick#2201] [Help] [answered]
I would like to change the RGB color in bulk, the code is as follows:

`mogrify -path input_dir -channel RGB -separate -swap 0,2 -combine output_dir`

       However, it replies that 'mogrify: unrecognized colorspace `6' @error/mogrify.c/MogrifyImageCommand/4511/'.

       It works if I just deal with one figure. Looking forward to your help.

Best regards
Yu

**Comment by dlemstra** [comment]
Is the `- separate` a typo are are you really using that instead of `-separate`

**Reply by Python-Zhou** [reply]
Thank you for your reply. It is the `-separate`, I have typed with a mistake in the Github.

**Comment by snibgo** [comment, marked as answer]
Your "input_dir" and "output_dir" are the wrong way round. The last argument should be the files, perhaps with a wildcard, that you want as input, eg "mydir/*.png". "-path" should be the output directory.

But there also seems to be a bug with "mogrify", or "magick mogrify" in IM v7.0.8-64. "-combine" should not need an argument. Try "-combine RGB" instead.

**Reply by Python-Zhou** [reply]
Thank you very much for your help.
